### PR TITLE
rsync: pass rsync_cv_SIGNED_CHAR_OK=yes to configure to bypass faulty signed char usability check

### LINKED ISSUE #317: delta transfer algorithm broken, since v3.2.4
Since v3.2.4 the delta transfer algorithm is broken - on some version<>architecture combinations.
As a result big files are transferred whole again, not only the changed part.

used test file with 4MB size:
```
dd if=/dev/urandom of=foobar0  bs=1M count=4
cp foobar0 foobar1
date >> foobar1
```
--> so foobar0 and foobar 1 are almost the same, except the appended date.

used rsync command lines:
```
rsync  --inplace --no-whole-file --partial --progress --archive --human-readable --stats --checksum foobar0 server:foobar0
rsync  --inplace --no-whole-file --partial --progress --archive --human-readable --stats --checksum foobar1 server:foobar1
```
--> on 2nd command is expected to transfer only the small changed part, but instead sometimes the whole file is transferred, logfile with -vvvv captured.

test result regarding delta transfer:

```
aarch64_(v3.2.3)--to--armv7l_(v3.2.4)	fail
aarch64_(v3.2.3)--to--x86_64_(v3.2.4)	ok

aarch64_(v3.2.4)--to--armv7l_(v3.2.4)	ok
aarch64_(v3.2.4)--to--x86_64_(v3.2.4)	fail

x86_64_(v3.2.4)--to--armv7l_(v3.2.4)	fail
x86_64_(v3.2.4)--to--x86_64_(v3.2.4)	ok

```
Building a x86_64_(v3.2.4) version with --disable-roll-simd does not solve the issue.
I'm happy to provide more info if required, or performing another test-case
OS is archlinux and archlinuxarm for this test setup.

[aarch64_(v3.2.3)--to--armv7l_(v3.2.4).txt](https://github.com/WayneD/rsync/files/8645141/aarch64_.v3.2.3.--to--armv7l_.v3.2.4.txt)
[aarch64_(v3.2.3)--to--x86_64_(v3.2.4).txt](https://github.com/WayneD/rsync/files/8645142/aarch64_.v3.2.3.--to--x86_64_.v3.2.4.txt)
[aarch64_(v3.2.4)--to--armv7l_(v3.2.4).txt](https://github.com/WayneD/rsync/files/8645144/aarch64_.v3.2.4.--to--armv7l_.v3.2.4.txt)
[aarch64_(v3.2.4)--to--x86_64_(v3.2.4).txt](https://github.com/WayneD/rsync/files/8645145/aarch64_.v3.2.4.--to--x86_64_.v3.2.4.txt)
[x86_64_(v3.2.4)--to--armv7l_(v3.2.4).txt](https://github.com/WayneD/rsync/files/8645146/x86_64_.v3.2.4.--to--armv7l_.v3.2.4.txt)
[x86_64_(v3.2.4)--to--x86_64_(v3.2.4).txt](https://github.com/WayneD/rsync/files/8645147/x86_64_.v3.2.4.--to--x86_64_.v3.2.4.txt)
[aarch64_(v3.2.3).txt](https://github.com/WayneD/rsync/files/8645148/aarch64_.v3.2.3.txt)
[aarch64_(v3.2.4).txt](https://github.com/WayneD/rsync/files/8645149/aarch64_.v3.2.4.txt)
[x86_64_(v3.2.4).txt](https://github.com/WayneD/rsync/files/8645150/x86_64_.v3.2.4.txt)
.

### PR DESCRIPTION
**Pull request by ismail**
Fixes https://github.com/WayneD/rsync/issues/317

### CODE REVIEW

**Comment by ismail** [comment]
Updated commit message.

**Comment by ismail** [comment]
> Please provide a better issue/pull request title and/or description!

Done.

**Comment by ismail** [comment]
@chenrui333 Could I kindly ask for a review?

File: Formula/rsync.rb [hunk 1]
```diff
@@ -52,6 +60,11 @@ def install
     # function multiversioning on older versions of macOS
     args << "--disable-simd" if MacOS.version < :catalina
 
+    # Remove in the next release along with autoconf/automake deps
+    # Needed for rsync-schar-fix-3.2.4.diff
+    system "autoreconf", "-fi"
+    system "autoconf", "-o", "configure.sh"
```
Comment (SMillerDev) [inline Formula/rsync.rb:66]: This is only needed on ARM right?

File: Formula/rsync.rb [hunk 2]
```diff
@@ -38,6 +40,12 @@ class Rsync < Formula
     apply "patches/fileflags.diff"
   end
 
+  patch do
+    url "https://download.samba.org/pub/rsync/rsync-schar-fix-3.2.4.diff"
+    sha256 "1c3baba01f3087ae2d2aa9195eedf97adf2b57f0482e9435a0433525c399635f"
+    apply "rsync-schar-fix-3.2.4.diff"
```
Comment (SMillerDev) [inline Formula/rsync.rb:46]: No need to specify this since it's the same as the filename

File: Formula/rsync.rb (hunk 1 above)
Comment (ismail) [reply Formula/rsync.rb:66]: Yes indeed, but patch is also correct on x86.

Comment (SMillerDev) [reply Formula/rsync.rb:65]: Seems better to only patch what is broken, yes.

**Comment by ismail** [comment]
I've now changed the fix to basically bypass the faulty configure check as suggested by @fxcoudert, this makes things easier than making the patch arch-specific.

**Comment by ismail** [comment]
Manually confirmed that the check is now successfully bypassed:

```
rsync-20220629-64800-1nrh0df/rsync-3.2.4/config.h
723:#define SIGNED_CHAR_OK 1
```

@SMillerDev Please re-review, thank you! :)

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**